internal/server/server.go

```diff
@@ -238,6 +240,16 @@ func (s *Server) Run(ctx context.Context) error {
 	return err
 }
 
+func runTelemetryHeartbeat(ctx context.Context, tel *Telemetry) error {
+	waiter := repeat.NewWaiter(backoff.NewConstantBackOff(time.Hour))
+	for {
+		tel.EnqueueHeartbeat()
+		if err := waiter.Wait(ctx); err != nil {
+			return err
```

Comment:
Yes, that's right. `waiter.Wait` only returns an error when the context is cancelled (or when the backoff returns Stop, but the constant backoff used here never returns Stop). We want this goroutine to exit when the context is cancelled, that's the signal the process is being shutdown. 

If one of the goroutines in the `errgroup` return a non-nil error, it will cancel the context, which indicates to the other goroutines that they should exit as well.

Reply:
I forgot to say, this is the same behaviour we had before. When the context was cancelled the goroutine would exit in `repeat.Start`.